VMAccess/vmaccess.py

```diff
@@ -302,25 +322,44 @@ def _set_sshd_config(config, name, val):
             # Match block must be put in the end of sshd config
             break
     if notfound:
-        config.insert(i, "{0} {1}".format(name, val))
+        if i is None:
+            i = 0
+            config.insert(i, "{0} {1}".format(name, val))
     return config
 
 
+def _get_default_ssh_config_filename():
+    os_name = ext_utils.get_line_starting_with("NAME", constants.os_release)
+    if os_name is not None:
+        # the default ssh config files are present
+        # /var/lib/waagent/Microsoft.OSTCExtensions.VMAccessForLinux-<version>/resources/
+        if re.match("centos", os_name, re.IGNORECASE):
+            return "centos_default"
+        if re.match("debian", os_name, re.IGNORECASE):
+            return "debian_default"
+        if re.match("fedora", os_name, re.IGNORECASE):
+            return "fedora_default"
+        if re.match("red\s?hat", os_name, re.IGNORECASE):
+            return "redhat_default"
+        if re.match("suse", os_name, re.IGNORECASE):
+            return "SuSE_default"
+        if re.match("ubuntu", os_name, re.IGNORECASE):
+            return "ubuntu_default"
+        return "default"
```

Comment:
Curious how this list is compiled, is VMAccess not supported on other distros?
For example, the [`osutil` factory](https://github.com/Azure/WALinuxAgent/blob/develop/azurelinuxagent/common/osutil/factory.py) in WALinuxAgent has more distros not mentioned here like `arch`, `kali`, etc.

Reply:
Any for distro not mentioned here, we don't ship a default ssh configuration file. We then default to the default-default ssh configuration